### ISSUE TITLE
Create v5 API, freeze v4, deprecate v3

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#11273]
We want to do an API version bump. The idea is to:
- [ ] Rename v4dev to v5. v5 will become the new "experimental / in progress" API version
  -  [ ] Move addons-code-manager to v5 instead of v4dev
- [ ] Freeze v4 API
- [ ] Deprecate v3 API
  - [ ] Move Firefox to v4 endpoints (investigate differences that will impact Firefox, file bugs to get them to switch).

In addition, the following needs to happen but does not block *right now* (will come back to bite us if we don't act on it though):
- [ ] Move addons-frontend to use v5. Blocked by https://github.com/mozilla/addons-frontend/issues/6422 essentially.

**Comment by eviljeff** [comment]
> Move addons-frontend to use v5 ... does not block right now 

It doesn't block creating v5, but it practically does block freezing v4 because we won't then be able to add any new feature to AMO frontend that relies on an API change (i.e. almost every new feature).  If v4 isn't frozen it makes moving Firefox to v4 more risky too.

**Comment by diox** [comment]
Yeah. Technically we can plan for this because AFAIK there are no features planned for the frontend in Q2 or Q3 that would require extra API work besides [recommended add-ons frontend changes](https://github.com/orgs/mozilla/projects/69) in Q2. But it would definitely makes things a lot easier to just migrate frontend to v5 first, you're right.

**Comment by eviljeff** [comment]
A recent shower thought was to have a `/api/vFirefox70/` (or whatever version of Firefox it was first introduced in) and only expose the subset of endpoints needed by Firefox at that point to `/vFirefox70/` to de-couple the whole v3 (v4, v5...) API from what Firefox needs.  If we were going to the effort of having changes made in Firefox I'd be in favour of jumping straight to a v5 base too 

The blockers on removal are still getting consumers to upgrade though.  We need an agreed commitment to move off the existing API versions - and that includes addons-frontend (so an agreement within the team) - backed by goals to do the work.

**Comment by eviljeff** [comment]
Updated the issue with new issues/bugs.  We need to do some sort of external communications too (not a massive rush if we're looking at a 10 month timescale, but the sooner then better obviously)

**Comment by eviljeff** [comment]
https://docs.google.com/document/d/1I0C56T8uK-IZ3_h2NiasOkkyjxzzx2SMF7rNXAHpvvU/edit# for comms

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDSRV-11
